src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,66 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.InvalidUserAgeException;
+import core.basesyntax.exceptions.InvalidUserDataException;
+import core.basesyntax.exceptions.InvalidUserLoginException;
+import core.basesyntax.exceptions.InvalidUserPasswordException;
+import core.basesyntax.exceptions.NullUserException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new NullUserException();
+        } else if (isUserValid(user)) {
```

Comment:
The method `isUserValid` should not be wrapped in a try-catch block for exceptions that are meant to be thrown. Instead, let the exceptions propagate naturally, as they are already being handled in the `register` method.